hicexplorer/hicPlotAverageRegions.py

```diff
@@ -77,33 +77,31 @@ def main(args=None):
 
     matrix = matrix.toarray()
     matrix = np.triu(matrix)
-    matrix = rotate(matrix, 45)
+    matrix = rotate(matrix, 45, cval=np.nan)
     matrix_shapes = matrix.shape
     matrix = matrix[:matrix_shapes[0] // 2, :]
-    norm = None
-    if args.vMax is not None or args.vMin is not None:
-        matrix = matrix.clip(min=args.vMin, max=args.vMax)
-    if args.log1p or args.log:
```

Comment:
Please bring the clearing back. It is there to avoid unexpected results in plotting. 
1. Zero for log is not defined and creates a `RuntimeWarning: divide by zero encountered in log`
2. Nan and Inf stay Nan and Inf after applied log. This can create unwanted scaling in the plots.

Reply:
I understand what you say but, I am not sure it is valid:
In your tests, the matrix you are using has negative values. In the current "clearing", you do not test if the matrix has negative values, thus, in the test:
https://github.com/deeptools/HiCExplorer/blob/047f2a1fc3f37ffa852ecafd4ba7a8b648118c03/hicexplorer/test/general/test_hicPlotAverageRegions.py#L29-L36
What is happening with your "clearing" is that:
all the values which are 0, nan or infinite will receive the negative value.
Then you apply a LogScale with negative values, they become nan and thus 'white bins':
![defaults_log](https://user-images.githubusercontent.com/30404086/73935215-4ffbd600-48e0-11ea-9951-77a9654f65a3.png)
And there is no error.
Also, with the current clearing, if you have nan values and 0 values, the nan values will become 0 because with `np.nanmin(matrix[mask_nan == False])` you do the minimum outside nan values but not outside 0 values and the infinite values also...
I understand your point that matplotlib is unstable but I think it is better to have nice plot for the moment rather than having plots which are misleading with all invalid values with the same color as the minimum and having the corner the same color as the matrix like in:

![defaults](https://user-images.githubusercontent.com/30404086/73844719-64c56480-4821-11ea-98ba-3c94a67e3902.png)

I think if you want a clearing for log of negative values or log of infinite, we can put them to nan, something like:
```
    if args.log1p or args.log:
        if args.log1p:
            matrix += 1

        mask = matrix <= 0
        mask_inf = np.isinf(matrix)

        try:
            matrix[mask] = np.nan
            matrix[mask_inf] = np.nan
        except Exception:
            log.warning('Clearing of matrix failed. Plotting can fail.')
```

Reply:
What if the matrix has negative values? I think i am missing something here. What i mean is that if plotting is going to be done over a log ratio matrix for example.

Reply:
@LeilyR  this is only when log or log1p options are set.

Reply:
We have negative values in the matrix? This should not happen... and log is also not defined for the class of real numbers. I do see we need to catch it, but there should not be any negative numbers present.

Reply:
There is no negative value in the original matrix but after rotation, there is...

Reply:
Maybe it would be safer to do like in pgt (pcolormesh_45deg)

Reply:
What is sure it that there is negative values before "Clearing".

Reply:
I need to check the rotation, that it is creating negative values seems not right to me.

Reply:
For me this function does really weird things. When you rotate an image of pretty high resolution, maybe it does not matter so much but here. People may want to draw conclusions on the bins.
Here is a minimal example with strange results:
```
>>> import numpy as np
>>> from scipy.sparse import load_npz
>>> from scipy.ndimage import rotate
>>> matrix = [[1, 0, 0, 0, 0],
...           [0, 1, 1, 0, 0],
...           [0, 1, 1, 0, 0],
...           [0, 0, 0, 2, 0],
...           [0, 0, 0, 0, 1]]
>>> matrix = np.array(matrix, dtype = 'float')
>>> rotate(matrix, 45)
array([[ 0.        ,  0.        ,  0.        ,  0.        ,  0.        ,  0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.        , -0.00838727,  0.        ,  0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.49035765,  0.12712625, -0.07614408,  0.        ,  0.        ],
       [ 0.        ,  0.47933733,  1.45516654,  1.        ,  1.60775943,  0.95555026,  0.        ],
       [ 0.        ,  0.        ,  0.49035765,  0.12712625, -0.07614408,  0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.        , -0.00838727,  0.        ,  0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.        ,  0.        ,  0.        ,  0.        ,  0.        ]])
```
```

Reply:
You are right... It seems the tools is not working as I was expecting it. Thanks a lot for pointing it out.